File: nfs2024_matlab/simpleFlowSolver2D/applyBCV.m
% # applyBCV.m ##########################################
% Programm: 		    	simpleFlowSolver2D
% Content:					apply boundary conditions
% Author:                   S.H. Tey, June 2024
% Email:                    sheu.tey@stud.uni-due.de
% Matriculation Number:     3117568
% -------------------------------------------------------
% functions for boundary condition applications
%    - application of periodic boundary condition to
%      to both x, y directions
% =======================================================

function Phi = applyBCV(Phi) % ##########################
% ------- parameters ------------------------------------
global Ifi Ifim Ila Ilap Jfi Jfim Jla Jlap
% ------- operattions -----------------------------------
% Dirichlet on all sides
Phi(Ifim, Jfi:Jla) = 0;
Phi(Ilap, Jfi:Jla) = 0;
Phi(Ifi:Ila, Jfim) = 0;
Phi(Ifi:Ila, Jlap) = 0;
end % ###################################################

File: nfs2024/simpleFlowSolver2D/applyBCV.jl
function applyBCV!(Phi)
    """
        apply ... boundary condition to pressure
    """
    nG = 1
    ImaAll, JmaAll = size(Phi)
	Ifi = nG+1; Ifip = Ifi+1; Ifim = Ifi-1
	Ila = ImaAll - nG; Ilap = Ila+1; Ilam = Ila-1
	Jfi = nG+1; Jfip = Jfi+1; Jfim = Jfi-1
	Jla = JmaAll - nG; Jlap = Jla+1; Jlam = Jla-1
    
    Phi[Ifim, Jfi:Jla] .= 0
    Phi[Ilap, Jfi:Jla] .= 0
    Phi[Ifi:Ila, Jfim] .= 0
    Phi[Ifi:Ila, Jlap] .= 0
end